pilot/pkg/bootstrap/server.go

```diff
@@ -162,14 +163,16 @@ type Server struct {
 	mixerSAN         []string
 	kubeClient       kubernetes.Interface
 	startFuncs       []startFunc
-	clusterStore     *clusterregistry.ClusterStore
 	httpServer       *http.Server
 	grpcServer       *grpc.Server
 	secureGRPCServer *grpc.Server
 	discoveryService *envoy.DiscoveryService
 	istioConfigStore model.IstioConfigStore
 	mux              *http.ServeMux
-	kubeRegistry     *kube.Controller
+	kubeRegistry     map[string]*kube.Controller
```

Comment:
why add this map since we already have Controller.registries?

Reply:
I do not understand your comment.  There is no references to "Controller.registries" in the current code.   This is needed to have multiple kubeRegistries on the pilot server function.  Previously these new controllers were created in the secret controller code,  but now with a common library need these controllers hanging off the pilot server object.